### PR TITLE
Simplify note manager

### PR DESCRIPTION
**Pull request by mlists**
The note manager was doing quite a lot. This splits off a shooter state machine which can now cease execution. The intake component has taken over the role of the injector once more.

### CODE REVIEW

File: controllers/note.py [hunk 1]
```diff
@@ -1,50 +1,39 @@
...
 
-    chassis: ChassisComponent
-    injector_component: InjectorComponent
     intake: IntakeComponent
 
     shot_desired = will_reset_to(False)
-    intake_desired = will_reset_to(False)
-    intake_cancel_desired = will_reset_to(False)
+
+    def __init__(self):
```
Comment (outsidermm) [inline controllers/note.py:15]: Type hint for method 
```
__init__
try_intake
cancel_intake
try_shoot
has_note
execute
holding_note
not_holding_note
```

File: controllers/note.py [hunk 2]
```diff
@@ -1,50 +1,39 @@
...
-        return self.injector_component.has_note()
-
-    def translation_to_goal(self) -> Translation2d:
-        return (
-            get_goal_speaker_position().toTranslation2d()
-            - self.chassis.get_pose().translation()
-        )
+        return self.intake.has_note()
 
     def has_just_fired(self) -> bool:
```
Comment (outsidermm) [inline controllers/note.py:31]: Feedback decorator?

File: controllers/shooter.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            - self.chassis.get_pose().translation()
+        )
+
+    @default_state
+    def idling(self):
+        self.shooter_component.set_range(self.translation_to_goal().norm())
+
+    @state(first=True)
+    def aiming(self):
+        self.shooter_component.set_range(self.translation_to_goal().norm())
```
Comment (outsidermm) [inline controllers/shooter.py:31]: What is the point of setting the range in line 31 when it’s going to be updated again in line 41

File: controllers/shooter.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    intake: IntakeComponent
+
+    def translation_to_goal(self) -> Translation2d:
+        return (
+            get_goal_speaker_position().toTranslation2d()
+            - self.chassis.get_pose().translation()
+        )
+
+    @default_state
+    def idling(self):
```
Comment (outsidermm) [inline controllers/shooter.py:26]: Add type hint for method:
```
Idling
aiming
Firing 
```

File: components/intake.py [hunk 5]
```diff
@@ -140,10 +152,16 @@ def retract(self) -> None:
...
     def intake(self) -> None:
         self.direction = self.Direction.FORWARD
+        self.desired_injector_speed = (
+            0.0 if self.has_note() else self.inject_intake_speed
+        )
 
     def outtake(self) -> None:
         self.direction = self.Direction.BACKWARD
 
+    def inject(self) -> None:
```
Comment (auscompgeek) [inline components/intake.py:162]: nit: It might be confusing to call a method on the intake `inject`. Maybe `feed_shooter` will be clearer for the team.

**Review by auscompgeek — APPROVED** [review]
LGTM

**Pull request merged**